src/main/java/appeng/client/me/ItemRepo.java

```diff
@@ -160,43 +163,46 @@ public void updateView()
 				continue;
 			}
 
-			if( viewMode == ViewItems.CRAFTABLE )
-			{
-				is = is.copy();
-				is.setStackSize( 0 );
-			}
-
 			if( viewMode == ViewItems.STORED && is.getStackSize() == 0 )
 			{
 				continue;
 			}
 
 			final String dspName = searchMod ? Platform.getModId( is ) : Platform.getItemDisplayName( is );
+			boolean foundMatchingItemStack = false;
 			notDone = true;
 
 			if( m.matcher( dspName.toLowerCase() ).find() )
 			{
-				this.view.add( is );
 				notDone = false;
+				foundMatchingItemStack = true;
 			}
 
 			if( terminalSearchToolTips && notDone && !searchMod )
 			{
-				for( final Object lp : Platform.getTooltip( is ) )
+				final List<String> tooltip = Platform.getTooltip( is );
+
+				for( final String line : tooltip )
 				{
-					if( lp instanceof String && m.matcher( (CharSequence) lp ).find() )
+					if( m.matcher( line ).find() )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)